module-check/src/main/clojure/clojure/core/typed/lex_env.clj

```diff
@@ -7,10 +7,12 @@
 
 (alter-meta! *ns* assoc :skip-wiki true)
 
-(def lex-env? (con/hash-c? con/local-sym? r/Type?))
+(def lex-env? (con/hash-c? con/local-sym? (some-fn r/Type? r/Unique?)))
 (def prop-set? (con/set-c? fr/Filter?))
 (def alias-env? (con/hash-c? con/local-sym? obj/RObject?))
 
+(defonce ^:dynamic *used-locals* (atom {} :validator ((con/hash-c? symbol? con/boolean?) (some-fn delay? r/Type? r/Unique?))))
```

Comment:
You should play around with atom validators and the `hash-c?` functions until they become more familiar. These are not correct.

I want `*used-locals*` to be a set of symbols. This way we only write to them once per local, instead of twice.